src/qt/walletmodel.cpp

```diff
@@ -756,3 +779,296 @@ bool WalletModel::getDefaultWalletRbf() const
+    for (std::map<std::string, NameNewReturn>::iterator i = pendingNameFirstUpdate.begin();
+         i != pendingNameFirstUpdate.end(); )
+    {
+        JSONRPCRequest jsonRequest;
+        UniValue params1(UniValue::VOBJ);
+        UniValue res1, val;
+        // hold the error returned from name_firstupdate, via
+        // completePendingNameFirstUpdate, or empty on success
+        // this will drive the error-handling popup
+        std::string completedResult;
+
+        std::string name = i->first;
+        std::string txid = i->second.hex;
+        std::string rand = i->second.rand;
+        std::string data = i->second.data;
+        std::string toaddress = i->second.toaddress;
+
+        params1.push_back (Pair("txid", txid));
+        jsonRequest.params = params1;
+        // if we're here, the names doesn't exist
+        // should we remove it from the DB?
+        try {
+            res1 = gettransaction( jsonRequest);
+        }
+        catch (const UniValue& e) {
+            UniValue message = find_value( e, "message");
+            std::string errorStr = message.get_str();
+            LogPrintf ("gettransaction error for name %s: %s\n",
+                       name.c_str(), errorStr.c_str());
+            ++i;
+            continue;
+        }
+
+        val = find_value (res1, "confirmations");
+        if (!val.isNum ())
+        {
+            LogPrintf ("No confirmations for name %s\n", name.c_str());
+            ++i;
+            continue;
+        }
+
+        const int confirms = val.get_int ();
+        LogPrintf ("Pending Name FirstUpdate Confirms: %d\n", confirms);
+
+        if ( confirms < 12)
+        {
+            ++i;
+            continue;
+        }
+
+        if(getEncryptionStatus() == Locked)
+        {
+            if (QMessageBox::Yes != QMessageBox::question(this,
+                  tr("Confirm wallet unlock"),
+                  tr("Namecoin Core is about to finalize your name registration for name <b>%1</b>, by sending name_firstupdate. If your wallet is locked, you will be prompted to unlock it. Pressing cancel will delay your name registration by one block, at which point you will be prompted again. Would you like to proceed?").arg(QString::fromStdString(name)),
+                  QMessageBox::Yes|QMessageBox::Cancel,
+                  QMessageBox::Cancel))
+            {
+                LogPrintf ("User cancelled wallet unlock pre-name_firstupdate. Waiting 1 block.\n");
+                return;
+            }
+
+            LogPrintf ("Attempting wallet unlock ...\n");
+            WalletModel::UnlockContext ctx(this->requestUnlock ());
+            if (!ctx.isValid ())
+            {
+                return;
+            }
+            else
+            {
+                // NOTE: the reason we're doing this here and not at the same
+                // time as the one beneath it is because when ctx is destroyed
+                // (subsequently after leaving this block) the wallet is locked.
```

Comment:
I don't like the duplicated code (even if it is just one line).  A way to get around the wallet unlock thing is the following:

```
std::unique_ptr<UnlockContext> unlock_ctx;
if (wallet locked) {
  ...
  unlock_ctx.reset(new UnlockContext());
}
this->completePendingNameFirstUpdate();
unlock_ctx.reset();
```